src/foam/lib/json/AnyKeyParser.java

```diff
@@ -10,12 +10,12 @@
 
 public class AnyKeyParser extends ProxyParser {
   public AnyKeyParser() {
-    super(new Alt(
-                  new Seq1(1,
-                           new Literal("\""),
-                           new Substring(new Repeat0(new NotChars("\""))),
-                           new Literal("\"")),
-                  new Seq1(0,
-                           new Substring(new Repeat0(new NotChars(" :{}"))))));
+    super(new Alt(new StringParser(), 
+                  new Substring(
+                    new Seq0(
+                      new Alt(new Range('A', 'z'), new Literal("_"), new Literal("$")), 
```

Comment:
Any way to extract this (ex. PermittedCharsParser) without creating a whole new file? (Repeated on line 19).